examples/large_models/utils/test_llm_streaming_response.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+import random
+import threading
+from queue import Queue
+
+import orjson
+import requests
+
+
+class Predictor(threading.Thread):
+    def __init__(self, args, queue):
+        threading.Thread.__init__(self)
+        self.args = args
+        self.queue = queue
+
+    def run(self):
+        for _ in range(self.args.num_requests_per_thread):
+            self._predict()
+
+    def _predict(self):
+        payload = self._format_payload()
+        with requests.post(self._get_url(), json=payload, stream=True) as response:
+            combined_text = ""
+            for chunk in response.iter_content(chunk_size=None):
+                if chunk:
+                    data = orjson.loads(chunk)
+                    combined_text += data["text"]
+        self.queue.put_nowait(f"payload={payload}\n, output={combined_text}\n")
+
+    def _get_url(self):
+        return f"http://localhost:8080/predictions/{self.args.model}"
+
+    def _format_payload(self):
+        prompt = _load_curl_like_data(self.args.prompt_text)
+        prompt_list = prompt.split(" ")
+        rp = len(prompt_list)
+        rt = self.args.max_tokens
+        if self.args.prompt_randomize:
+            rp = random.randint(1, len(prompt_list))
+            rt = random.randint(10, self.args.max_tokens)
+        cur_prompt_list = prompt_list[:rp]
+        cur_prompt = " ".join(cur_prompt_list)
+        return {
+            "prompt": cur_prompt,
+            "max_new_tokens": rt,
+        }
+
+
+def _load_curl_like_data(text):
+    """
+    Either use the passed string or load from a file if the string is `@filename`
+    """
+    if text.startswith("@"):
+        try:
+            with open(text[1:], "r") as f:
+                return f.read()
+        except Exception as e:
+            raise ValueError(f"Failed to read file {text[1:]}") from e
+    else:
+        return text
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "-m",
+        "--model",
+        type=str,
+        help="The model to use for generating text. If not specified we will pick the first model from the service as returned by /v1/models",
```

Comment:
does this need a default value or will ts with no model in url automatically return the prediction from first model?